### ISSUE TITLE
Changing interface to type for ActionCreator<P> in 3.0-beta-2

### ISSUE THREAD

**Issue by MistyKuu** [aikoven/typescript-fsa#70]
I'm currently using typescript-fsa version 2.5. I'm augmenting ActionCreator interface like this:

```
declare module "typescript-fsa" {
  export interface ActionCreator<P> {
    type: string;
    payloadType: P;
    match: (action: Fsa.AnyAction) => action is Action<P>;
    (payload: P, meta?: Meta): Action<P>;
  }
}
```

The reason I'm doing it is that I extended it with payloadType. In 3.0 I'm not able to do that since types in typescript cannot be augmented. Is there any reason why you decided to use type?

**Comment by QuantumInformation** [comment]
is this lib still maintained?

**Comment by aikoven** [comment]
@QuantumInformation Yes!

@MistyKuu That's because in 3.0 the `ActionCreator` type is composed of two types using intersection (`&`), which lets us use different signature when `Payload` type is `void`. However, with newer TypeScript versions this is no longer needed, because you can skip void parameters (see https://github.com/Microsoft/TypeScript/pull/27522).

There are a few other things I'd like to fix before releasing 3.0 stable, let's keep this open until then.

For now, you can safely continue to use 2.5.

And by the way, if what you're trying to achieve with this augmentation is extracting payload type from action creator type, you can use conditional types for that.

**Comment by QuantumInformation** [comment]
Cool thanks @aikoven 

We are seeing this:
<img width="1143" alt="Screenshot 2019-03-12 at 07 57 24" src="https://user-images.githubusercontent.com/216566/54183633-83e63580-449c-11e9-887b-5d820ee61705.png">

**Comment by QuantumInformation** [comment]
This happened after moving to the new native create react app --typescripe

**Comment by aikoven** [comment]
@QuantumInformation Can you show the definition of `SaveIssue` action creator?

**Comment by QuantumInformation** [comment]
`export const SaveIssue = actionCreator.async<null, IssueDetails , string>('SaveIssue');`

**Comment by QuantumInformation** [comment]
If I turn off     "strict": true,
no error.

**Comment by QuantumInformation** [comment]
Fixed it updating the generic type

export const LoadIssue = actionCreator.async< null, IssueDetails , string>('LoadIssue');
to
export const LoadIssue = actionCreator.async< {issueId: number}, IssueDetails , string>('LoadIssue');

**Comment by QuantumInformation** [comment]
Have you thought of setting up a discord server for this project, or adding it to reactiflux?

**Comment by aikoven** [comment]
> Have you thought of setting up a discord server for this project, or adding it to reactiflux?

There's not so much activity on this project. And honestly, I don't have much spare time to maintain a chat room.

**Comment by QuantumInformation** [comment]
ok, thx anyway

**Comment by aikoven** [comment]
@MistyKuu fixed in [`v3.0.0`](https://github.com/aikoven/typescript-fsa/releases/tag/v3.0.0).
